Jint/Runtime/TypeConverter.cs

```diff
@@ -522,6 +525,25 @@ public static void CheckObjectCoercible(Engine engine, JsValue o)
                     matchingByParameterCount = matchingByParameterCount ?? new List<Tuple<T, JsValue[]>>();
                     matchingByParameterCount.Add(new Tuple<T, JsValue[]>(m, arguments));
                 }
+                else if (parameterInfos.Length > arguments.Length)
+                {
+                    // check if we got enough default values to provide all parameters (or more in case some default values are provided/overwritten)
+                    var defaultValuesCount = parameterInfos.Count(x => x.HasDefaultValue);
+                    if (parameterInfos.Length <= arguments.Length + defaultValuesCount)
+                    {
+                        // create missing arguments from default values
+                        var argsWithDefaults = new List<JsValue>(arguments);
+                        for (var i = arguments.Length; i < parameterInfos.Length; i++)
+                        {
+                            var param = parameterInfos[i];
+                            var value = JsValue.FromObject(_engine, param.DefaultValue);
+                            argsWithDefaults.Add(value);
+                        }
+
+                        matchingByParameterCount = matchingByParameterCount ?? new List<Tuple<T, JsValue[]>>();
```

Comment:
this needs to be `new System.Collections.Generic.List<Tuple<T, JsValue[]>>();` as it now breaks compilation due to Esprima having also the List<T>

Reply:
Added full qualified namespace to my List call, but not to the lower one, as it is a plain copy of code a few lines above. Since I can't reproduce the issue of not compiling, I hope that fixes it.

Reply:
After rebasing the describes problem was obvious...